compute_tools/src/compute.rs

```diff
@@ -343,4 +345,70 @@ impl ComputeNode {
         self.prepare_pgdata()?;
         self.run()
     }
+
+    // Look for core dumps and collect backtraces.
+    //
+    // EKS worker nodes have following core dump settings:
+    //   /proc/sys/kernel/core_pattern -> core
+    //   /proc/sys/kernel/core_uses_pid -> 1
+    //   ulimint -c -> unlimited
+    // which results in core dumps being written to postgres data directory as core.<pid>.
+    //
+    // For local testing I also added support for lldb, which writes core dumps to /cores.
+    fn check_for_core_dumps(&self) {
+        let core_dump_dir = match std::env::consts::OS {
+            "macos" => Path::new("/cores/"),
+            _ => Path::new(&self.pgdata),
+        };
+
+        info!(
+            "checking for core dumps in {}",
+            core_dump_dir.to_string_lossy()
+        );
+
+        let paths = fs::read_dir(core_dump_dir).expect("cannot list pgdata for core dumps");
+        for path in paths {
+            if let Some(name) = path
+                .expect("can't read pgdata directory entry")
+                .path()
+                .file_name()
+            {
+                let name = name.to_string_lossy();
+                if name.starts_with("core.") {
+                    warn!("core dump found: {}, collecting backtrace", name);
+                    let full_core_path = std::path::Path::new(core_dump_dir).join(name.as_ref());
+                    let full_core_path = full_core_path.to_string_lossy();
```

Comment:
At your discretion:

```rust
let files = fs::read_dir("/tmp")?;
let cores = files.filter_map(|x| {
    let entry = x.ok()?;
    let _ = entry.path().to_str()?.prefix("core.")?;
    Some(entry)
});

for entry in cores {
  let path = entry.path();
  ...
}
```

Reply:
Thanks, I didn't like that part as well. Will rewrite.